shepherd_bms/src/calcs.cpp

```diff
@@ -148,6 +148,34 @@ void calcContDCL(AccumulatorData_t *bmsdata)
     }
 }
 
+void calcCCL(AccumulatorData_t *bmsdata)
+{
+    int16_t currentLimit = 0x7FFF;
+
+    for(uint8_t c = 0; c < NUM_CHIPS; c++)
+    {
+        for(uint8_t cell = 0; cell < NUM_CELLS_PER_CHIP; cell++)
+        {   
+            // Apply equation
+            uint16_t tmpCCL = (MAX_VOLT - ((bmsdata->chipData[c].openCellVoltage[cell] + VOLT_SAG_MARGIN) * 10000)) / (bmsdata->chipData[c].cellResistance[cell] * 10);
+            // Multiplying resistance by 10 to convert from mOhm to Ohm and then to Ohm * 10000 to account for the voltage units
+
+            //Taking the minimum DCL of all the cells
+            if(tmpCCL < currentLimit) currentLimit = tmpCCL;
+        }
+    }
+
+    // ceiling for current limit
+    if (currentLimit > MAX_CELL_CURR) 
+    {
+        bmsdata->dischargeLimit = MAX_CELL_CURR;
+    }
+    else 
+    {
+        bmsdata->dischargeLimit = currentLimit;
+    }
+}
```

Comment:
calcs.cpp is going to be deleted and is essentially a copy of analyzer, its probably not a big deal to keep it here but may just make sense to delete from here first